gluten-core/src/main/scala/org/apache/spark/sql/execution/ColumnarCollapseTransformStages.scala

```diff
@@ -19,13 +19,58 @@ package org.apache.spark.sql.execution
 import io.glutenproject.GlutenConfig
 import io.glutenproject.backendsapi.BackendsApiManager
 import io.glutenproject.execution._
+import io.glutenproject.metrics.MetricsUpdater
+import io.glutenproject.substrait.SubstraitContext
+import io.glutenproject.substrait.rel.RelBuilder
 
+import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.expressions.{Attribute, SortOrder}
+import org.apache.spark.sql.catalyst.plans.physical.Partitioning
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.metric.SQLMetric
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
 import java.util.concurrent.atomic.AtomicInteger
 
+import scala.collection.JavaConverters._
+
+/**
+ * A bridge to connect [[SparkPlan]] and [[TransformSupport]] and provide the substrait plan
+ * `ReadRel` for the child columnar iterator, so that the [[TransformSupport]] always has input. It
+ * would be transformed to `ValueStreamNode` at native side.
+ */
+case class InputIteratorTransformer(child: SparkPlan) extends UnaryTransformSupport {
```

Comment:
Should we combine `InputIteratorTransformer` and `ColumnarInputAdapter` later on?

Reply:
It seems we do not need `ColumnarInputAdapter` any more